src/sidebar/markdown-commands.js

```diff
@@ -239,18 +241,26 @@ function transformLines(state, start, end, callback) {
  * Toggle Markdown-style formatting around a block of text.
  *
  * @param {EditorState} state - The current state of the input field.
- * @param {string} prefix - The prefix to add or remove before each line
- *                          of the selection.
+ * @param {string|((lineNum: number) => string)} prefixBuilder - The prefix
+ *        to add or remove before each line of the selection, or a callback
+ *        which returns said prefix for a specific line number (useful for
+ *        multi-line selections that require different prefixes per line).
  * @return {EditorState} - The new state of the input field.
  */
-export function toggleBlockStyle(state, prefix) {
+export function toggleBlockStyle(state, prefixBuilder) {
   const start = state.selectionStart;
   const end = state.selectionEnd;
+  /** @param {number} lineNum */
+  const prefixToString = lineNum =>
+    typeof prefixBuilder === 'function'
+      ? prefixBuilder(lineNum)
+      : prefixBuilder;
 
   // Test whether all lines in the selected range already have the style
   // applied
   let blockHasStyle = true;
-  transformLines(state, start, end, (state, lineStart) => {
+  transformLines(state, start, end, (state, lineStart, _, lineNum) => {
```

Comment:
I'm not sure what's usually your take when it comes to naming unused params (I'm talking about the third one, `_`).

I have considered changing the signature, so that the callback expectes an object and you can decide what properties to unpack: `({ state, lineStart, lineNum }) => {...}`. However, that would require changes in more places.

Ultimatelly, we could just make it explicit and name it `lineEnd`, even if we don't use it.

Reply:
> I'm not sure what's usually your take when it comes to naming unused params (I'm talking about the third one, _).

It hasn't come up all that often. If there are many parameters, I would use an object. In this instance there aren't that many though. In JS you can just omit names entirely IIRC (`(state, lineStart,, lineNum)`) but this feels prone to misreading. I've been working in Rust recently which has a built-in convention of using `_<name>` (eg. `_lineEnd`), that I quite like. It's simple and hopefully we can configure tools (eg. ESLint) to ignore such params.

Reply:
> eg. `_lineEnd`

This is the approach I've used and seen other use in other projects. It gives a clear name and states that the param is not used in one sight.